src/main/java/GameRunner.java

```diff
@@ -1,24 +1,51 @@
 public class GameRunner {
 
-    private Game game = new Game();
-    private Cli cli = new Cli(System.in, System.out, game);
+    private Game game;
+    private Cli cli = new Cli(System.in, System.out);
+
+    public GameRunner(Game game) {
+        this.game = game;
+    }
 
     public void runGame() {
-        cli.askGameType();
+        cli.askGameMode();
         String choice = cli.takeInput();
 
         if (choice.equals("1")) {
-            System.out.println("Human");
-        } else {
-            while (!game.isBoardFull()) {
-                cli.showBoard();
-                cli.askForInput();
-                String input = cli.takeInput();
-                int squareNumber = game.convertInputToSquareNumber(input);
-                game.setSquareToXMark(squareNumber);
-                if (squareNumber + 1 < game.getSquares().length) game.setSquareToO(squareNumber + 1);
+            while (gameOngoing()) {
+                runTurn(game.getActivePlayer());
+                game.alternatePlayer();
             }
-            cli.showBoard();
+        }
+//        else {
+//            while (!game.isBoardFull()) {
+//                cli.displayBoard(game.getSquares());
+//                cli.askInput(game.getPlayerOne());
+//                String input = cli.takeInput();
+//                int squareNumber = game.convertInputToSquareNumber(input);
+//                game.setSquareMark(squareNumber, "X");
+//                if (squareNumber + 1 < game.getSquares().length) game.setSquareMark(squareNumber + 1, "O");
+//            }
+//            cli.displayBoard(game.getSquares());
+//        }
+    }
+
+    public boolean gameOngoing() {
+        return !game.isBoardFull(game.getGrid()) && !game.isGameWon(game.getGrid());
+    }
+
+    public void runTurn(Player activePlayer) {
+        cli.displayBoard(game.getGrid().getSquares());
+        String input = cli.askAndTakeInput(activePlayer);
+        int squareNumber = game.convertInputToSquareNumber(input);
+        game.markSquare(activePlayer, game.getGrid(), squareNumber);
```

Comment:
All args to markSquare are methods of game - can't this method just take the input?